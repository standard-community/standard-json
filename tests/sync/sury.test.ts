import S from "sury";
import { describe, expect, it } from "vitest";

import { loadVendor, toJsonSchema } from "~/index.js";
import type { ToJsonSchemaFn } from "~/vendors/utils.js";

describe("sync check", () => {
  it("sury", () => {
    // Loading the vendor
    loadVendor("sury", S.toJSONSchema as ToJsonSchemaFn);

    const schema = S.schema({
      myString: S.string,
      myUnion: S.union([S.number, S.boolean]),
    }).with(S.meta, { description: "My neat object schema" });

    const jsonSchema = toJsonSchema.sync(schema);
    expect(jsonSchema).toMatchSnapshot();
  });
});
