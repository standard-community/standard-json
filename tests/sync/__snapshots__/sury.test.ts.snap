// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sync check > sury 1`] = `
{
  "additionalProperties": true,
  "description": "My neat object schema",
  "properties": {
    "myString": {
      "type": "string",
    },
    "myUnion": {
      "anyOf": [
        {
          "type": "number",
        },
        {
          "type": "boolean",
        },
      ],
    },
  },
  "required": [
    "myString",
    "myUnion",
  ],
  "type": "object",
}
`;
